plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -638,6 +667,20 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 				   fn, error_local->message);
 			return TRUE;
 		}
+#endif
+	}
+
+#ifdef __FreeBSD__
+	if (!uefi_supported) {
+		g_error_free (error_local);
+		error_local = NULL;
+
+		/* check kenv for UEFI information */
+		uefi_supported = fu_plugin_uefi_capsule_kenv_enabled (&error_local);
+	}
+#endif
+
+	if (!uefi_supported) {
```

Comment:
I think this new flow is really hard to follow with ifdef in the else blocks.  Can we please Make some static functions, or bring `fu_plugin_uefi_capsule_kenv_enabled` into `fu_plugin_uefi_capsule_smbios_enabled` perhaps?  It seems to me like this `fu_plugin_startup` could avoid all changes if you put all the BSD logic into that function

Reply:
I think splitting it out as fu-uefi-capsule-common-linux.c and fu-uefi-capsule-common-freebsd.c might make this a lot easier to maintain. I can help on Monday with this if that would be useful.

Reply:
@hughsie, new files with functions that look something like this?
```c
gboolean fu_uefi_platform_startup_extra(void); // Linux: nothing, FreeBSD: check kenv
gboolean fu_uefi_platform_startup_fallback(void); // Linux: ignore smbios failure, FreeBSD: nothing
gboolean fu_uefi_platform_coldplug_extra(void); // Linux: check efivarsfs, FreeBSD: nothing
```

Reply:
My proposal would be something like https://github.com/fwupd/fwupd/pull/3318 -- I've merged the obviously correct easy ones directly to master thanks.